From: Alexander Graf <agraf@suse.de>
Date: Weg, 20 Mar 2013 15:17:25 +0100
Subject: [PATCH] omapdrm: fix compile with dma_addr_t > u32
Patch-Mainline: No

The omap_fb module passes a pointer to a u32 field to a function that expects a
dma_addr_t * parameter. Make the field dma_addr_t instead - it's an address after
all.

Fixes the following breakage:

  drivers/gpu/drm/omapdrm/omap_fb.c:231:6: error: passing argument 5 of 'omap_gem_rotated_paddr' from incompatible pointer type [-Werror]


Signed-off-by: Alexander Graf <agraf@suse.de>

Index: current/include/video/omapdss.h
===================================================================
--- current.orig/include/video/omapdss.h
+++ current/include/video/omapdss.h
@@ -389,8 +389,8 @@ struct omap_dss_cpr_coefs {
 };
 
 struct omap_overlay_info {
-	u32 paddr;
-	u32 p_uv_addr;  /* for NV12 format */
+	dma_addr_t paddr;
+	dma_addr_t p_uv_addr;  /* for NV12 format */
 	u16 screen_width;
 	u16 width;
 	u16 height;
Index: current/drivers/gpu/drm/omapdrm/Makefile
===================================================================
--- current.orig/drivers/gpu/drm/omapdrm/Makefile
+++ current/drivers/gpu/drm/omapdrm/Makefile
@@ -3,7 +3,7 @@
 # Direct Rendering Infrastructure (DRI)
 #
 
-ccflags-y := -Iinclude/drm -Werror
+ccflags-y := -Iinclude/drm
 omapdrm-y := omap_drv.o \
 	omap_irq.o \
 	omap_debugfs.o \
